code/lib/cli/src/automigrate/fixes/prompt-remove-react.ts

```diff
@@ -0,0 +1,133 @@
+
+        return {
+          files: (
+            await glob(slash(absolutePattern), {
+              ...commonGlobOptions(absolutePattern),
+              follow: true,
+            })
+          ).map((f) => relative(absoluteDirectory, f)),
+          directory,
+          titlePrefix,
+        };
+      })
+    )
+  ).reduce<boolean>((acc, { files }, i) => {
+    const filteredEntries = files.filter((s) => !s.endsWith('.mdx'));
+    if (filteredEntries.length < files.length) {
+      return true;
+    }
+    return acc;
+  }, false);
+  return result;
+}
+
+export const removeReactDependency: Fix<Options> = {
+  id: 'remove-react-dependency',
+
+  async check({ packageManager, mainConfig, storybookVersion, configDir }) {
+    let hasMDX = false;
+    let hasEssentials = false;
+    let hasDocs = false;
+
+    // when the user is using the react renderer, we should not prompt them to remove react
+    const frameworkPackageName = getFrameworkPackageName(mainConfig);
+    if (frameworkPackageName?.includes('react') || frameworkPackageName?.includes('nextjs')) {
+      return null;
+    }
+
+    // if the user has no dependency on react, we can skip this fix
+    const packageJson = await packageManager.retrievePackageJson();
+    if (
+      !packageJson?.dependencies?.['react'] &&
+      !packageJson?.peerDependencies?.['react'] &&
+      !packageJson?.devDependencies?.['react']
+    ) {
+      return null;
+    }
+
+    // do not prompt to remove react for older versions of storybook
+    if (!semver.gte(storybookVersion, '8.0.0')) {
+      return null;
+    }
+
+    const { addons } = mainConfig;
+    hasEssentials = !!(
+      addons &&
+      addons.find((addon) =>
+        typeof addon === 'string'
+          ? addon.endsWith('@storybook/addon-essentials')
+          : addon.name.endsWith('@storybook/addon-essentials')
+      )
+    );
+    hasDocs = !!(
+      addons &&
+      addons.find((addon) =>
+        typeof addon === 'string'
+          ? addon.endsWith('@storybook/addon-docs')
+          : addon.name.endsWith('@storybook/addon-docs')
+      )
+    );
+
+    hasMDX = !!(await detectMDXEntries(mainConfig.stories, configDir || process.cwd()));
+
+    return {
+      hasMDX,
+      hasEssentials,
+      hasDocs,
+    };
+  },
+  prompt({ hasMDX, hasDocs, hasEssentials }) {
+    const addons = [hasEssentials ? 'essentials' : '', hasDocs ? 'docs' : ''].filter(Boolean);
+    const addonReasonText =
+      addons.length > 0 ? `, because you are using ${addons.join(' and ')}` : '';
+
+    const start = dedent`
+      We detected that your project has a dependency for "react" that it might not need.
+      Nothing breaks by having it, you can safely ignore this message, if you wish.
+
+      Storybook asked you to add "react" as a direct dependency in the past${addonReasonText}.
+      However, since version 8.0, Storybook no longer requires you to provide "react" as a dependency.
```

Comment:
```suggestion
      We detected that your project has a dependency for "react" that it might not need.
      Nothing breaks by having it, you can safely ignore this message, if you wish.

      Storybook asked you to add "react" as a direct dependency in the past${addonReasonText}.
      However, since version 8.0, Storybook no longer requires you to provide "react" as a dependency.
      Some community addons might still wrongfully list "react" and "react-dom" as required peer dependencies, but since Storybook 7.6 it should not be needed in the majority of cases.
```